### DISCUSSION TITLE
Only use one buffer / reuse current buffer when I open new files?

### DISCUSSION THREAD

**Discussion by StealthOrc** [LazyVim/LazyVim#2630] [Q&A]
### How do I configure lazyvim so whenever I open a new file I only have it use one buffer, as in re use the currently open buffer?

I don't want to use \<leader\>bd or \<leader\>bo (bdelete) all the time to get rid of them and rather only work with **one main buffer**.
This happens when opening files from within Neo-Tree as well as telescope buffers (e.g.: \<leader\>space).

I found this issue on telescopes end that was realized in v0.15 it seems, but Im not sure..
issue: https://github.com/nvim-telescope/telescope.nvim/issues/2188/

in that issue they talked about re using buffers with already open files, but id like only one main buffer to exist and be reused.
Im quite new to Neovim still so any help would be appreciated!

**Comment by dusty-phillips** [comment]
Buffers represent open files in memory, so “reusing” them isn’t really a concept that makes sense. You are looking for automatically closing buffers when new ones open.

If it’s just a display question, the easiest option you have would be to disable the bufferline plugin. The buffers would be still be open, in neovim, but you wouldn’t see them on the bar. They would still be visible in the telescope open buffer picker and neotree buffer list, though.

You could also try something like https://github.com/chrisgrieser/nvim-early-retirement or https://github.com/axkirillov/hbac.nvim

Another thing you could experiment with is an autocommand that issues the space b o command whenever a file opens. No idea how well that would work though. 

Otherwise you’d be looking to overwrite the confirm keybindings in neotree and telescope to call a close all buffers command before opening the new one.

**Reply by StealthOrc** [reply]
Hmm the thing with just disabling bufferline and only making this a visual issue just seems kind of a workaround only, as the file would still be open and locked for other applications.. (for my setup on my personal setup this would most likely not be an issue though: ) 
I think a solution of the current buffer to be closed via an autocommand seems like a way to go maybe.. (in case I can still preservere/access the needed buffer Id of the to-be-closed-buffer in the autocommand.)

Thank you very much, I think this already nudged me in a good direction!
I'll take a look into it!

**Reply by LogicKahanHai** [reply]
Hello @StealthOrc! Did you find any solution to this? I was also looking for a way to do the same. Thanks!

**Reply by StealthOrc** [reply]
I did not look into it further sadly, sorry :/

**Reply by LogicKahanHai** [reply]
Oh alright! thanks, will look into it. If you want I can revert here, if I find a solution. :)

**Reply by StealthOrc** [reply]
Sure would be glad if you do!
